### PR TITLE
Use string-based attribute storage

### PR DESCRIPTION
**Pull request by Axel-Naumann**
But store only attributes that are overriding the style values.

### CODE REVIEW

**Comment by linev** [comment]
Now it compiles and will work after merging https://github.com/root-project/root/pull/3682

File: graf2d/gpadv7/v7/src/RDrawingOptsBase.cxx [hunk 1]
```diff
@@ -28,16 +28,19 @@
 std::shared_ptr<ROOT::Experimental::RDrawingAttrHolder> &
 ROOT::Experimental::RDrawingOptsBase::GetHolder()
 {
-   if (!fHolder)
+   if (!fHolder) {
       fHolder = std::make_shared<RDrawingAttrHolder>();
```
Comment (Axel-Naumann) [inline graf2d/gpadv7/v7/src/RDrawingOptsBase.cxx]: Don't we need a symmetrical "if I/O has read `fHolderIO`, set `fHolder` from it"?

File: tutorials/v7/draw_rh1.cxx [hunk 2]
```diff
@@ -45,8 +45,13 @@ void draw_rh1() {
 
    // Create a canvas to be displayed.
    auto canvas = RCanvas::Create("Canvas Title");
-   canvas->Draw(pHist)->Line().SetColor(RColor::kRed);
-   canvas->Draw(pHist2)->Line().SetColor(RColor::kBlue);
+   auto draw1 = canvas->Draw(pHist);
+   draw1->Line().SetColor(RColor::kRed);
+   draw1->Line().SetWidth(2);
```
Comment (Axel-Naumann) [inline tutorials/v7/draw_rh1.cxx]: Or `draw1->Line().SetColor(RColor::kRed).SetWidth(2);`

File: tutorials/v7/draw_rh1.cxx [hunk 3]
```diff
@@ -45,8 +45,13 @@ void draw_rh1() {
 
    // Create a canvas to be displayed.
    auto canvas = RCanvas::Create("Canvas Title");
-   canvas->Draw(pHist)->Line().SetColor(RColor::kRed);
-   canvas->Draw(pHist2)->Line().SetColor(RColor::kBlue);
+   auto draw1 = canvas->Draw(pHist);
```
Comment (Axel-Naumann) [inline tutorials/v7/draw_rh1.cxx]: Should we rename `draw` to `opts`?

File: js/scripts/JSRootPainter.v7.js [hunk 4]
```diff
@@ -4262,6 +4262,23 @@
       return col;
    }
 
+   TCanvasPainter.prototype.GetNewOpt = function(opts, name) {
+      if (!opts || !opts.fHolderIO || !name) return;
+
+      var map = opts.fHolderIO.fAttrNameVals;
+      if (!map || !map.length) return;
+
+      for (var i=0; i<map.length; ++i)
```
Comment (Axel-Naumann) [inline js/scripts/JSRootPainter.v7.js]: Here, a JS map https://stackoverflow.com/questions/1208222/how-to-do-associative-array-hashing-in-javascript could make things *much* faster...

File: graf2d/gpadv7/v7/src/RDrawingOptsBase.cxx (hunk 1 above)
Comment (linev) [reply graf2d/gpadv7/v7/src/RDrawingOptsBase.cxx]: Current  `temprorary` solution works **only** for storage of attributes in JSON. 

There is no way to correctly restore such object from ROOT or JSON file - shared_ptr/weak_ptr problems.

File: tutorials/v7/draw_rh1.cxx (hunk 2 above)
Comment (linev) [reply tutorials/v7/draw_rh1.cxx]: Grate, if it works this way! Examples will be change many times anyway.

File: js/scripts/JSRootPainter.v7.js (hunk 4 above)
Comment (linev) [reply js/scripts/JSRootPainter.v7.js]: Yes, but this requires creation of such JSON object already on C++ side.
In principle - possible, but not principal right now.
Much more important - play-through all scenarios with CSS usage

File: tutorials/v7/draw_rh1.cxx (hunk 3 above)
Comment (linev) [reply tutorials/v7/draw_rh1.cxx]: With such syntax `Line().SetColor(RColor::kRed).SetWidth(2)` one could skip variable completely.

File: js/scripts/JSRootPainter.v7.js (hunk 4 above)
Comment (linev) [reply js/scripts/JSRootPainter.v7.js]: Here is PR #3701 which optionally enable such feature for selected C++ objects/data members. Potentially can be enabled for attributes storage in the future

**Pull request merged**